### PR TITLE
feat: DTOSS-3025-Update-Data-Model

### PR DESCRIPTION
**Pull request by Warren-Pitterson**
## Description

Data Model has been confirmed so tables have been changed accordingly and existing functionality has been changed to match. 

The Database scripts will have to be run again going forward the current codebase references old tables and properties

## Context

New tables, with data types has been sent over and so the codebase need to match 

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ x] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: application/CohortManager/Set-up/database/create_statement.sql [hunk 1]
```diff
@@ -1,221 +1,451 @@
...
+    CREATE TABLE dbo.VALIDATION_EXCEPTION
+    (
+        VALIDATION_EXCEPTION_ID INT IDENTITY(1, 1) NOT NULL,
+        RULE_ID INT NOT NULL,
+        RULE_NAME NVARCHAR(200) NOT NULL,
+        WORKFLOW NVARCHAR(50) NOT NULL,
+        NHS_NUMBER BIGINT NOT NULL,
+        DATE_CREATED DATETIME NOT NULL,
+        CONSTRAINT PK_VALIDATION_EXCEPTION
+            PRIMARY KEY (VALIDATION_EXCEPTION_ID)
```
Comment (alex-clayton-1) [inline application/CohortManager/Set-up/database/create_statement.sql:169]: This is the old validation exception table structure

Comment (Warren-Pitterson) [reply application/CohortManager/Set-up/database/create_statement.sql]: I've just pulled the correct table from main, this should be correct now.

File: tests/screeningDataServicesTests/DemographicDataServiceTests/demographicDataServiceTests.cs [hunk 2]
```diff
@@ -48,6 +48,7 @@ public async Task Run_return_DemographicDataSaved_OK()
         var json = JsonSerializer.Serialize(_participant);
         var sut = new DemographicDataService(_logger.Object, _createResponse.Object, _createDemographicData.Object);
 
+        SetupRequest(json);
         SetupRequest(json);
```
Comment (alex-clayton-1) [inline tests/screeningDataServicesTests/DemographicDataServiceTests/demographicDataServiceTests.cs:52]: Duplicated line of code

Comment (Warren-Pitterson) [reply tests/screeningDataServicesTests/DemographicDataServiceTests/demographicDataServiceTests.cs]: removed

File: application/CohortManager/src/Functions/Shared/Model/Demographic.cs [hunk 3]
```diff
@@ -1,35 +1,44 @@
...
-    public string HomeAddressLine2 { get; set; }
-    public string HomeAddressLine3 { get; set; }
-    public string HomeAddressCity { get; set; }
-    public string HomeAddressPostcode { get; set; }
-    public string HomePhoneNumber { get; set; }
-    public string HomeEmailAddress { get; set; }
-    public string HomePhoneTextphone { get; set; }
-    public string EmergencyContactPhoneNumber { get; set; }
+    public string? ParticipantId { get; set; }
+    public string? NHSId { get; set; }
```
Comment (alex-clayton-1) [inline application/CohortManager/src/Functions/Shared/Model/Demographic.cs:8]: I'm wondering whether we should renamed this to NhsNumber for consistency

File: application/CohortManager/src/Functions/Shared/Model/Participant.cs [hunk 4]
```diff
@@ -7,7 +7,7 @@ public class Participant
     public string? RecordType { get; set; }
     public string? ChangeTimeStamp { get; set; }
     public string? SerialChangeNumber { get; set; }
-    public string NHSId { get; set; }
+    public string? NHSId { get; set; }
```
Comment (alex-clayton-1) [inline application/CohortManager/src/Functions/Shared/Model/Participant.cs:10]: Same as above

File: application/CohortManager/src/Functions/Shared/Model/Demographic.cs (hunk 3 above)
Comment (Warren-Pitterson) [reply application/CohortManager/src/Functions/Shared/Model/Demographic.cs]: renamed for Participant, Demographic and all references throughout the project for consistency

File: application/CohortManager/src/Functions/Shared/Model/Participant.cs (hunk 4 above)
Comment (Warren-Pitterson) [reply application/CohortManager/src/Functions/Shared/Model/Participant.cs]: renamed for Participant, Demographic and all references throughout the project for consistency

**Review by alex-clayton-1 — APPROVED** [review]

**Pull request merged**